### PR TITLE
Fixed update authorities for employee.

### PR DESCRIPTION
**Pull request by NKorzhik**
Fixed update authorities for employee.
## CHECK LIST
- [x]  СI passed
- [x]  Сode coverage >=95%
- [x]  PR is reviewed manually again (to make sure you have 100% ready code)
- [x]  All reviewers agreed to merge the PR
- [x]  I've checked new feature as logged in and logged out user if needed
- [x]  PR meets all conventions

### CODE REVIEW

File: service/src/main/java/greencity/service/ubs/UBSClientServiceImpl.java [hunk 1]
```diff
@@ -1952,10 +1953,14 @@ public Set<String> getAllAuthorities(String email) {
...
     @Override
-    public void updateEmployeesAuthorities(UserEmployeeAuthorityDto dto, String email) {
+    public void updateEmployeesAuthorities(UserEmployeeAuthorityDto dto) {
         Employee employee = employeeRepository.findByEmail(dto.getEmployeeEmail())
             .orElseThrow(() -> new NotFoundException(EMPLOYEE_DOESNT_EXIST));
-        userRemoteClient.updateEmployeesAuthorities(dto, email);
+        try {
+            userRemoteClient.updateEmployeesAuthorities(dto);
+        } catch (HystrixRuntimeException e) {
+            throw new BadRequestException(EMPLOYEE_DOESNT_EXIST);
```
Comment (RomaMocherniuk) [inline service/src/main/java/greencity/service/ubs/UBSClientServiceImpl.java]: I think we should check the response core and throw `BadRequestException(EMPLOYEE_DOESNT_EXIST)` only if the response code is `404`

File: service/src/test/java/greencity/service/ubs/UBSClientServiceImplTest.java [hunk 2]
```diff
@@ -2420,10 +2431,9 @@ void updateEmployeesAuthoritiesService() {
 
         when(employeeRepository.findByEmail(dto.getEmployeeEmail())).thenReturn(Optional.of(ModelUtils.getEmployee()));
 
-        ubsService.updateEmployeesAuthorities(dto, dto.getEmployeeEmail());
+        ubsService.updateEmployeesAuthorities(dto);
 
-        verify(userRemoteClient, times(1)).updateEmployeesAuthorities(
-            dto, "test@mail.com");
+        verify(userRemoteClient, times(1)).updateEmployeesAuthorities(dto);
```
Comment (RomaMocherniuk) [inline service/src/test/java/greencity/service/ubs/UBSClientServiceImplTest.java:2453]: `verify` w/o second param means `times(1)`

File: service/src/test/java/greencity/service/ubs/UBSClientServiceImplTest.java [hunk 3]
```diff
@@ -2395,9 +2396,19 @@ void getAllAuthorities() {
     @Test
     void updateEmployeesAuthorities() {
         UserEmployeeAuthorityDto dto = ModelUtils.getUserEmployeeAuthorityDto();
-        userRemoteClient.updateEmployeesAuthorities(dto, "test@mail.com");
-        verify(userRemoteClient, times(1)).updateEmployeesAuthorities(
-            dto, "test@mail.com");
+        userRemoteClient.updateEmployeesAuthorities(dto);
+        verify(userRemoteClient, times(1)).updateEmployeesAuthorities(dto);
```
Comment (RomaMocherniuk) [inline service/src/test/java/greencity/service/ubs/UBSClientServiceImplTest.java:2428]: same here

File: service-api/src/main/java/greencity/dto/employee/EmployeeDto.java [hunk 4]
```diff
@@ -18,10 +18,10 @@ public class EmployeeDto {
     @Min(1)
     private Long id;
     @NotNull
-    @Pattern(regexp = "[ЁёІіЇїҐґЄєА-Яа-яA-Za-z-'\\s.]{1,30}")
+    @Pattern(regexp = "^(?!\\.)(?!.*\\.$)(?!.*?\\.\\.)(?!.*?\\-\\-)(?!.*?\\'\\')[-'ʼ ґҐіІєЄїЇА-Яа-я+\\w.]{1,30}$")
     private String firstName;
     @NotNull
-    @Pattern(regexp = "[ЁёІіЇїҐґЄєА-Яа-яA-Za-z-'\\s.]{1,30}")
+    @Pattern(regexp = "^(?!\\.)(?!.*\\.$)(?!.*?\\.\\.)(?!.*?\\-\\-)(?!.*?\\'\\')[-'ʼ ґҐіІєЄїЇА-Яа-я+\\w.]{1,30}$")
```
Comment (RomaMocherniuk) [inline service-api/src/main/java/greencity/dto/employee/EmployeeDto.java]: could you please add tests for there regexes

File: core/src/main/java/greencity/controller/ManagementEmployeeController.java [hunk 5]
```diff
@@ -285,9 +283,8 @@ public ResponseEntity<Object> getAllAuthorities(@RequestParam String email) {
     })
     @PreAuthorize("@preAuthorizer.hasAuthority('EDIT_EMPLOYEES_AUTHORITIES', authentication)")
     @PutMapping("/edit-authorities")
-    public ResponseEntity<Object> editAuthorities(@Valid @RequestBody UserEmployeeAuthorityDto dto,
-        @ApiIgnore Principal principal) {
-        ubsClientService.updateEmployeesAuthorities(dto, principal.getName());
+    public ResponseEntity<Object> editAuthorities(@Valid @RequestBody UserEmployeeAuthorityDto dto) {
```
Comment (RomaMocherniuk) [inline core/src/main/java/greencity/controller/ManagementEmployeeController.java:288]: please uopdate ApiResponses
This edpoint does not return  `response = UserEmployeeAuthorityDto.class`

Comment (RomaMocherniuk) [reply core/src/main/java/greencity/controller/ManagementEmployeeController.java:288]: Could we have `404` response code in this case https://github.com/ita-social-projects/GreenCityUBS/pull/1007/files#diff-d5ba7e4400c6d3e88dfdd27f144c7c964c8819955aa261014b596806cd6b7ac9R1962 ?

File: service-api/src/main/java/greencity/dto/employee/EmployeeDto.java [hunk 6]
```diff
@@ -18,10 +18,10 @@ public class EmployeeDto {
     @Min(1)
     private Long id;
     @NotNull
-    @Pattern(regexp = "[ЁёІіЇїҐґЄєА-Яа-яA-Za-z-'\\s.]{1,30}")
+    @Pattern(regexp = "^(?!\\.)(?!.*\\.$)(?!.*?\\.\\.)(?!.*?\\-\\-)(?!.*?\\'\\')[-'ʼ ґҐіІєЄїЇА-Яа-я+\\w.]{1,30}$")
```
Comment (RomaMocherniuk) [inline service-api/src/main/java/greencity/dto/employee/EmployeeDto.java]: please move this regex to the constant

File: core/src/main/java/greencity/controller/ManagementEmployeeController.java (hunk 5 above)
Comment (NKorzhik) [reply core/src/main/java/greencity/controller/ManagementEmployeeController.java:288]: Done

File: service-api/src/main/java/greencity/dto/employee/EmployeeDto.java (hunk 6 above)
Comment (NKorzhik) [reply service-api/src/main/java/greencity/dto/employee/EmployeeDto.java]: done

File: service/src/main/java/greencity/service/ubs/UBSClientServiceImpl.java (hunk 1 above)
Comment (NKorzhik) [reply service/src/main/java/greencity/service/ubs/UBSClientServiceImpl.java]: done

File: service/src/test/java/greencity/service/ubs/UBSClientServiceImplTest.java (hunk 2 above)
Comment (NKorzhik) [reply service/src/test/java/greencity/service/ubs/UBSClientServiceImplTest.java:2453]: done

File: service/src/test/java/greencity/service/ubs/UBSClientServiceImplTest.java (hunk 3 above)
Comment (NKorzhik) [reply service/src/test/java/greencity/service/ubs/UBSClientServiceImplTest.java:2428]: done

File: service-api/src/main/java/greencity/dto/employee/EmployeeDto.java (hunk 4 above)
Comment (NKorzhik) [reply service-api/src/main/java/greencity/dto/employee/EmployeeDto.java]: done

**Review by sergoliarnik — APPROVED** [review]

**Review by JJJazl — APPROVED** [review]

**Review by juseti — APPROVED** [review]

**Pull request merged**